modules/nf-core/bamclipper/main.nf

```diff
@@ -0,0 +1,54 @@
+
+process BAMCLIPPER {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bamclipper:1.0.0--hdfd78af_2':
+        'biocontainers/bamclipper:1.0.0--hdfd78af_2' }"
+
+    input:
+    tuple val(meta), path(bam), path(bai), path(bed)
+
+    output:
+    tuple val(meta), path("*.primerclipped.bam")    , emit: bam
+    tuple val(meta), path("*.primerclipped.bam.bai"), emit: bai
+    path "versions.yml"                             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '1.0.0' // WARN: Version information not provided by tool on CLI
+    """
+    bamclipper.sh \\
+        -b ${bam} \\
+        -p ${bed} \\
+        -n $task.cpus \\
+        ${args}
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        bamclipper: $VERSION
+        samtools: \$( samtools --version |& sed '1!d ; s/samtools //' )
```

Comment:
I'm not sure if this is necessary

Reply:
What exactly? Returning the versions yml is for sure I think

Reply:
E.g. as in:
https://github.com/nf-core/modules/blob/7e56daae390ff896b292ddc70823447683a79936/modules/nf-core/fastqc/main.nf#L38

I think this was also mentioned in the comments I had to delete.

Reply:
If the comment was about samtools versioning, particularly the samtools version was the vital aspect to get the containers rights. Not a bad thing to know the version for reproducibility.

Reply:
ah, yes, sorry - I meant the samtools version!

yes, agreed!